components/operator-service/src/main/java/com/wso2telco/dep/operatorservice/service/BlackListWhiteListService.java

```diff
@@ -56,68 +59,62 @@ public BlackListWhiteListService()
-		MSISDNValidationDTO msisdnValidationDTO = new MSISDNValidationDTO();
-
-		final String apiID_ = dto.getApiID();
-		final String apiName_ = dto.getApiName();
-		final String userId_ = dto.getUserID();
-
-
-			try{
-
-				msisdnValidationDTO.setValid(Arrays.asList(msisdns));
-				msisdnValidationDTO.setValidationRegex(dto.getValidationRegex());
-				msisdnValidationDTO.setValidationPrefixGroup(dto.getValidationPrefixGroup());
-				msisdnValidationDTO.setValidationDigitsGroup(dto.getValidationDigitsGroup());
-				msisdnValidationDTO.process();
+		List<String> valid = new ArrayList<>();
+		List<String> invalid = new ArrayList<>();
+		APIBlacklistWhitelistResponseDTO responseDTO = new APIBlacklistWhitelistResponseDTO(0,0);
+		int success = 0;
 
+		Matcher matcher;
+		try {
 
-				// load already black listed numbers
-				List<MsisdnDTO> alreadyBlacklisted;
-
-				alreadyBlacklisted = dao.getBlacklisted(apiID_);
-
-				// Remove already black listed from the list
-				for (MsisdnDTO msisdn : alreadyBlacklisted) {
-					msisdnValidationDTO.getValidProcessed().remove(msisdn);
+			for (String m : dto.getMsisdnList()) {
+				matcher = bwRegex.matcher(m);
+				if (matcher.matches()) {
+					valid.add(m);
+				} else {
+					invalid.add(m);
 				}
-
-			} catch (Exception e){
-				LOG.error("Error while getting already blacklisted users",e);
-				throw new BlacklistException(BlacklistException.BlacklistErrorType.INTERNAL_SERVER_ERROR);
 			}
 
-
-			if (msisdnValidationDTO.getValidProcessed().isEmpty()) {
-				LOG.debug(" All the numbers already black listed");
-				throw new BlacklistException(BlacklistException.BlacklistErrorType.USER_ALREADY_BLACKLISTED);
+			if (!valid.isEmpty()) {
+				dto.setMsisdnList(valid);
+				success = dao.blacklist(dto);
 			}
 
-		try {
-			dao.blacklist(msisdnValidationDTO, apiID_, apiName_, userId_);
+			responseDTO.setProcessed(success);
+			responseDTO.setFailed(invalid.size() + (valid.size() - success));
+
 		} catch (Exception e) {
-			LOG.error("blacklist ", e);
+			LOG.error("Unable to process blacklist", e);
 			throw new BusinessException(GenaralError.INTERNAL_SERVER_ERROR);
 		}
+
+		if(invalid.size() > 0){
+			LOG.error("Could not process due to invalid format: " + invalid);
+		}
+
+		return responseDTO;
     }
 
-	public String[] loadBlacklisted(MSISDNSearchDTO searchDTO) throws BusinessException {
+	public BlacklistWhitelistCountResponseDTO count(BlacklistWhitelistDTO dto) throws BusinessException {
 		try {
-			return dao.getBlacklisted(searchDTO);
+			return dao.getCount(dto);
 		} catch (Exception e) {
 			LOG.error("loadBlacklisted", e);
 			throw new NumberBlackListException(GenaralError.INTERNAL_SERVER_ERROR);
 		}
 	}
 
-	public void removeBlacklist(int apiId, final String userMSISDN) throws BusinessException {
+	public void remove(BlacklistWhitelistDTO dto) throws BusinessException {
 		try {
-			dao.removeBlacklist(apiId, userMSISDN);
+			Matcher matcher = bwRegex.matcher(dto.getMsisdnList().get(0));
```

Comment:
Will check whether the msisdn list is empty before accessing the 0th element. Since no validation has placed in the controller com.wso2telco.services.bw.Queries#removeFromBlacklist, there is a possibility of the list could be empty or null